tests/lib/scheduler_test.py

```diff
@@ -0,0 +1,105 @@
+# pylint: disable=redefined-outer-name,protected-access
+from datetime import datetime
+from unittest.mock import Mock, patch
+from uuid import uuid4
+
+import pytest
+from cloudformation_cli_python_lib.scheduler import CloudWatchScheduler
+
+from botocore.exceptions import ClientError
+
+
+@pytest.fixture
+def mock_boto3_session():
+    return Mock("cloudformation_cli_python_lib.scheduler.Session", autospec=True)()
+
+
+@pytest.fixture
+def mock_handler_request():
+    mock_request = Mock(
+        "cloudformation_cli_python_lib.utils.HandlerRequest", autospec=True
+    )()
+    mock_request.requestContext = {}
+    mock_request.serialize.return_value = {}
+    return mock_request
+
+
+def test_instantiates_boto3_client(mock_boto3_session):
+    cw_scheduler = CloudWatchScheduler(boto3_session=mock_boto3_session)
+    mock_boto3_session.client.assert_called_once_with(("events"))
+    assert cw_scheduler.client == mock_boto3_session.client.return_value
+
+
+@patch(
+    "cloudformation_cli_python_lib.scheduler.CloudWatchScheduler._min_to_cron",
+    return_value="cron('30 16 21 11 ? 2019')",
+)
+@patch(
+    "cloudformation_cli_python_lib.scheduler.uuid4", autospec=True, return_value=uuid4()
+)
+def test_reschedule_after_minutes(
+    mock_uuid4, mock_min_to_cron, mock_boto3_session, mock_handler_request
+):
```

Comment:
i'm not a big fan of the patch decorators, as they quickly degrade readability. generally, in the RPDK, we've been using `with` statements:

https://github.com/aws-cloudformation/aws-cloudformation-rpdk-python-plugin/blob/9eda207789e9d904a529dff802f5c73addacb2a7/tests/plugin/codegen_test.py#L87

because if those get too long, you can split them up:

https://github.com/aws-cloudformation/aws-cloudformation-rpdk-python-plugin/blob/9eda207789e9d904a529dff802f5c73addacb2a7/tests/plugin/codegen_test.py#L174-L181

it isn't super important; i know `@patch` is common, but getting away from that has helped keep the number of test parameter manageable, and also makes it much easier to parametrize tests.